File: MB-Chat/observability/trace-context.ts
import { createCorrelationId, createTraceId } from "./correlation";
import type { TraceContext } from "./types";

export function createTraceContext(input: {
  tenant_id: string;
  request_type: string;
  source_layer: string;
  parent_trace_id?: string;
  trace_id?: string;
  correlation_id?: string;
}): TraceContext {
  return {
    trace_id: input.trace_id ?? createTraceId(),
    correlation_id: input.correlation_id ?? createCorrelationId(),
    parent_trace_id: input.parent_trace_id,
    tenant_id: input.tenant_id,
    request_type: input.request_type,
    source_layer: input.source_layer,
    created_at: new Date().toISOString(),
  };
}

File: MB-Whatsapp/observability_py/trace_context.py
from datetime import UTC, datetime

from .correlation import create_correlation_id, create_trace_id
from .types import TraceContext


def create_trace_context(
    tenant_id: str,
    request_type: str,
    source_layer: str,
    parent_trace_id: str | None = None,
    trace_id: str | None = None,
    correlation_id: str | None = None,
) -> TraceContext:
    return TraceContext(
        trace_id=trace_id or create_trace_id(),
        correlation_id=correlation_id or create_correlation_id(),
        parent_trace_id=parent_trace_id,
        tenant_id=tenant_id,
        request_type=request_type,
        source_layer=source_layer,
        created_at=datetime.now(UTC).isoformat(),
    )
